magenta/music/musicxml_parser.py

```diff
@@ -0,0 +1,516 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""MusicXML parser.
+
+Simple MusicXML parser used to convert MusicXML
+into tensorflow.magenta.NoteSequence.
+"""
+
+# Imports
+# Python 2 uses integer division for integers. Using this gives the Python 3
+# behavior of producing a float when dividing integers
+from __future__ import division
+
+from zipfile import ZipFile
+import xml.etree.ElementTree as ET
+
+# Constants
+
+# Number of MIDI ticks per beat. Required by NoteSequence structure
+TICKS_PER_BEAT = 960
+
+# Global variables
+# Default to one division per measure
+current_divisions = 1
+
+# Default to a tempo of 120 beats per minute
+# (This is the default tempo according to the
+# Standard MIDI Files 1.0 Specification)
+current_tempo = 120
```

Comment:
This is value is mutable. It maintains the state of the currently identified tempo in the MusicXML file. However, there is a place where a default tempo is set if none is found; I have used this constant in that location. It is fixed in the next commit.